### PR TITLE
Add ability to pull information from tagged schemas

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Adds the ability to search and pull information about fields in an asdf file from the underlying schema files. For example, this enables one to store keywords pertaining to a given asdf object's data entries within the schema(s) defining that object. This however requires that the underlying schemas to all be tagged.

### CODE REVIEW

**Review by perrygreenfield — COMMENTED** [review]
Mainly a comment on the docstring and name of the node class.

File: asdf/_node_info.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                break
+
+            current_nodes = next_nodes
+            current_depth += 1
+
+        return root_info
+
+    def collect_info(self, preserve_list=True):
+        """
+        Collect the information from the NodeSchemaData tree, and return it as nested dict.
```
Comment (perrygreenfield) [inline asdf/_node_info.py:198]: This reference needs updating.

**Review by perrygreenfield — COMMENTED** [review]
Only one lingering name left.

File: asdf/_node_info.py (hunk 1 above)
Comment (WilliamJamieson) [reply asdf/_node_info.py]: Done

**Review by perrygreenfield — APPROVED** [review]
LGTM

**Pull request merged**